src/server/jobs/search-index-sync.ts

```diff
@@ -1,69 +1,48 @@
-import { createJob } from './job';
-import { dbWrite } from '~/server/db/client';
-import { client } from '~/server/meilisearch/client';
-import { ModelStatus } from '@prisma/client';
-import { EnqueuedTask } from 'meilisearch';
-
-export const searchIndexSync = createJob('search-index-sync', '33 4 * * *', async () => {
-  // Get all models and add to meilisearch index
-  if (!client) return;
-
-  const allTasks = await Promise.all([prepareModelIndex()]);
-});
-
-const READ_BATCH_SIZE = 1000;
-async function prepareModelIndex() {
-  if (!client) return;
-
-  let offset = 0;
-  const modelTasks: EnqueuedTask[] = [];
-  const allTasks: EnqueuedTask[] = [];
-  while (true) {
-    const models = await dbWrite.model.findMany({
-      skip: offset,
-      take: READ_BATCH_SIZE,
-      select: {
-        id: true,
-        name: true,
-        type: true,
-        nsfw: true,
-        poi: true,
-        checkpointType: true,
-        locked: true,
-        underAttack: true,
-        earlyAccessDeadline: true,
-        mode: true,
-        allowNoCredit: true,
-        allowCommercialUse: true,
-        allowDerivatives: true,
-        allowDifferentLicense: true,
-        userId: true,
+import { createJob, UNRUNNABLE_JOB_CRON } from './job';
+import * as searchIndex from '~/server/search-index';
+
+const searchIndexSets = {
+  models: searchIndex.modelsSearchIndex,
+  tags: searchIndex.tagsSearchIndex,
+  users: searchIndex.usersSearchIndex,
+  articles: searchIndex.articlesSearchIndex,
+  images: searchIndex.imagesSearchIndex,
+};
+
+export const searchIndexJobs = Object.entries(searchIndexSets)
+  .map(([name, searchIndexProcessor]) => [
+    createJob(
+      `search-index-sync-${name}`,
+      '*/30 * * * *',
```

Comment:
This might be too high - but wanted to play it extra safe. TBH, I expect we can run this every min to 5min.